### PR TITLE
RHOAI Dashboard Scale testing

### PR DESCRIPTION
**Pull request by smandaRH**

### CODE REVIEW

**Comment by kpouget** [comment]
@smandaRH , I don't know why the test isn't running. I'll investigate that

**Comment by kpouget** [comment]
on your side, you can look at the failed checks.
One is easy, just run this:
```
./run_toolbox.py repo generate_ansible_default_settings
```
the rest should be more or less self explicit. Let me know if you don't understand how to fix some issues :)

**Comment by kpouget** [comment]
/ok-to-test
Hello @smandaRH , you need to use this comment now to launch the test:
```
/test rhoai-light notebooks dashboard_scale_test
```

**Comment by smandaRH** [comment]
sure @kpouget, Thanks for letting me know.

**Comment by kpouget** [comment]
@smandaRH you need to rebase and resolve the merge conflict before you can test again

**Comment by kpouget** [comment]
```
```
<<<<<<< scale-test
    ./run_toolbox.py from_config cluster set_project_annotation --prefix sutest --suffix node_selector --extra "$dedicated"
    ./run_toolbox.py from_config cluster set_project_annotation --prefix sutest --suffix toleration --extra "$dedicated"

    ./run_toolbox.py rhods update_datasciencecluster --enable [dashboard,workbenches,datasciencepipelines]
=======
    ./run_toolbox.py rhods update_datasciencecluster --enable [dashboard,workbenches]
>>>>>>> main
```
you need to make sure that this looks like:
```
    if test_config clusters.sutest.compute.dedicated; then
        ./run_toolbox.py from_config cluster set_project_annotation --prefix sutest --suffix node_selector
        ./run_toolbox.py from_config cluster set_project_annotation --prefix sutest --suffix toleration
    fi

    ./run_toolbox.py rhods update_datasciencecluster --enable [dashboard,workbenches,datasciencepipelines]
```

**Comment by smandaRH** [comment]
sure @kpouget , i will rebase it and resolve the conflicts

**Comment by kpouget** [comment]
failure is a flake (=outside of our control). Retrying
/test rhoai-light notebooks dashboard_scale_test

**Comment by kpouget** [comment]
[flake](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/407/pull-ci-openshift-psap-topsail-main-rhoai-light/1805244140007460864/artifacts/light/prepare/artifacts/000__cluster__wait_fully_awake/), retesting after merging https://github.com/openshift-psap/topsail/pull/418
 /test rhoai-light notebooks dashboard_scale_test

**Comment by kpouget** [comment]
test [flaked](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/407/pull-ci-openshift-psap-topsail-main-rhoai-light/1805561874071162880/artifacts/light/prepare/artifacts/000__cluster__wait_fully_awake/_ansible.log).
Retesting after https://github.com/openshift-psap/topsail/pull/421 has been merged.

**Comment by kpouget** [comment]
Hello @smandaRH, 
when a test fails, can you try to:
* show where it failed
    * include a link to the most relevant file/directory  
* if you think it is a flake 
    * in this case you can relaunch the test
*  or what went wrong
    * in this case you push a commit to try to solve it before retrying
---

eg, here, the users failed at [this level](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/407/pull-ci-openshift-psap-topsail-main-rhoai-light/1805605233930276864/artifacts/light/test/artifacts/002__driver_notebooks__dashboard_scale_test/ods-ci/ods-ci-2/final_screenshot.png):

image
I don't think it's a flake, so we need to understand what's causing this failure and what to do to get rid of it!

**Comment by smandaRH** [comment]
sure @kpouget, in the previous run, I faced an internal error 
image
so tried again. I will make sure to mention it in a similar format mentioned above.

**Comment by kpouget** [comment]
👍🏻 
this error that you observed is indeed a flake, but a flake in RHOAI (and you forgot to include the link, so that we can easily go to the artifacts and study the context of the failure). If this happens often, we need
1. to make sure that RHOAI is aware of it
2. to try to have a workaround in our code, if it's blocking us

**Comment by kpouget** [comment]
there was a flake in your last test that I (try to) solve [there](https://github.com/openshift-psap/topsail/pull/440)

File: projects/cluster/roles/cluster_build_push_image/tasks/main.yml [hunk 1]
```diff
@@ -1,43 +1,22 @@
...
 
 - name: Create namespace for images if does not exist
-  shell:
-    set -o pipefail;
-    oc create namespace "{{ cluster_build_push_image_namespace }}"
-       --dry-run=client
-       -oyaml
-    | tee "{{ artifact_extra_logs_dir }}/src/000_namespace.yml"
-    | oc apply -f-
+  shell: set -o pipefail; oc create namespace "{{ cluster_build_push_image_namespace }}" --dry-run=client -oyaml | tee "{{ artifact_extra_logs_dir }}/src/000_namespace.yml" | oc apply -f-
```
Comment (kpouget) [inline projects/cluster/roles/cluster_build_push_image/tasks/main.yml:8]: @smandaRH , can you try to revert the modifications of this file?  they remove new lines, we don't really want that

Comment (smandaRH) [reply projects/cluster/roles/cluster_build_push_image/tasks/main.yml:8]: Sure, will modify the changes requested.

**Comment by smandaRH** [comment]
Running Test after updating

**Comment by smandaRH** [comment]
Running After updating SSL layer to the cluster Minio endpoint

File: projects/cluster/roles/cluster_deploy_minio_s3_server/files/deploy_minio.yaml [hunk 2]
```diff
@@ -119,38 +119,39 @@ spec:
...
+            [[ $retcode != 7 ]] && break
+            sleep 5 # Connection refused, server not ready
+          done
+          echo "Minio is running, configuring it ..."
 
            _mc() {
              mc --config-dir /tmp --insecure "$@"
            }
            MINIO_SERVER_NAME=minio
+
```
Comment (kpouget) [inline projects/cluster/roles/cluster_deploy_minio_s3_server/files/deploy_minio.yaml:136]: can you remove this empty line?

File: projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml [hunk 3]
```diff
@@ -117,43 +131,52 @@
...
-       mc --config-dir /tmp cp /etc/os-release minio/{{ notebooks_dashboard_scale_test_minio_bucket_name }}; # without it, cp may fail if the bucket is empty
+       mc --config-dir /tmp cp --insecure /etc/os-release minio/{{ notebooks_dashboard_scale_test_minio_bucket_name }}; # without it, cp may fail if the bucket is empty
   failed_when: false
 
+- name: Delete S3 secret minio
+  command: oc delete secret/minio -n {{ tester_namespace }} --ignore-not-found
+
+- name: Create minio Secret
+  shell: set -o pipefail; set -e; SECRET_KEY=$(cat "{{ notebooks_dashboard_scale_test_secret_properties_file }}" | grep user_password | cut -d= -f2); ADMIN_PASSWORD=$(cat "{{ notebooks_dashboard_scale_test_secret_properties_file }}" | grep admin_password | cut -d= -f2); oc create secret generic minio -n "{{ notebooks_dashboard_scale_test_namespace }}" "--from-literal=MINIO_ROOT_PASSWORD=$ADMIN_PASSWORD" "--from-literal=MINIO_S3_ACCESS_KEY=minio" "--from-literal=MINIO_S3_SECRET_KEY=$SECRET_KEY"
+
```
Comment (kpouget) [inline projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml:177]: can you reformat this into multiple lines? something like that:
```suggestion
  shell: |
      set -o pipefail; 
      set -e; 
      
      SECRET_KEY=$(cat "{{ notebooks_dashboard_scale_test_secret_properties_file }}" | grep user_password | cut -d= -f2); 
      ADMIN_PASSWORD=$(cat "{{ notebooks_dashboard_scale_test_secret_properties_file }}" | grep admin_password | cut -d= -f2); 
      
      oc create secret generic minio -n "{{ notebooks_dashboard_scale_test_namespace }}" "--from-literal=MINIO_ROOT_PASSWORD=$ADMIN_PASSWORD" "--from-literal=MINIO_S3_ACCESS_KEY=minio" "--from-literal=MINIO_S3_SECRET_KEY=$SECRET_KEY"

```

File: projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml [hunk 4]
```diff
@@ -184,35 +207,25 @@
 # ---
 
 - name: Create the RHODS test job
-  shell:
-    set -o pipefail;
-    cat "{{ artifact_extra_logs_dir }}/src/000_notebooks_dashboard_scale_test.yaml"
-      | sed 's/$JOB_CREATION_TIME/'$(date "+%Y-%m-%dT%H:%M:%SZ" --utc)'/'
-      | oc create -f- -n "{{ tester_namespace }}"
+  shell: set -o pipefail; cat "{{ artifact_extra_logs_dir }}/src/000_notebooks_dashboard_scale_test.yaml" | sed 's/$JOB_CREATION_TIME/'$(date "+%Y-%m-%dT%H:%M:%SZ" --utc)'/' | oc create -f- -n "{{ tester_namespace }}"
```
Comment (kpouget) [inline projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml:210]: Here and below, you need to revert the formatting changes, must have been done automatically by your editor :/

File: projects/notebooks/testing/config.yaml [hunk 5]
```diff
@@ -12,7 +12,7 @@ ci_presets:
     clusters.create.ocp.workers.type: m6i.2xlarge
     clusters.create.ocp.compute.type: m6i.2xlarge
     tests.notebooks.users.sleep_factor: 3.0
-    tests.notebooks.test_flavor: ods-ci
+    tests.notebooks.test_flavor: dashboard-scale-test
```
Comment (kpouget) [inline projects/notebooks/testing/config.yaml:15]: can you revert the changes of this file, and instead extend the existing preset, with something like that:
```
dashboard_scale_test:
  tests.notebooks.test_flavor: dashboard-scale-test
  tests.notebooks.minio_bucket_name: ds-pipeline-dspa
  tests.notebooks.users.count: 5
  tests.notebooks.capture_prom_db: false
  clusters.sutest.compute.dedicated: false
  clusters.driver.compute.dedicated: false

```

File: projects/notebooks/testing/images/Containerfile.s3_artifacts_exporter [hunk 6]
```diff
@@ -4,6 +4,7 @@ RUN dnf -y --quiet install https://dl.fedoraproject.org/pub/epel/epel-release-la
  && dnf install -y --quiet s3cmd gettext \
  && dnf clean all
 
+
```
Comment (kpouget) [inline projects/notebooks/testing/images/Containerfile.s3_artifacts_exporter:7]: can you remove this empty line

File: projects/notebooks/testing/prepare_sutest.sh [hunk 7]
```diff
@@ -1,5 +1,12 @@
 #! /bin/bash
 
+prepare_sutest_deploy_operator() {
+    switch_sutest_cluster
+
+    process_ctrl::run_in_bg \
+        ./run_toolbox.py  cluster deploy_operator --catalog=redhat-operators --manifest_name=openshift-pipelines-operator-rh --namespace=all
+}
+
```
Comment (kpouget) [inline projects/notebooks/testing/prepare_sutest.sh:9]: this can be removed

File: projects/notebooks/testing/prepare_sutest.sh [hunk 8]
```diff
@@ -267,6 +275,18 @@ sutest_cleanup() {
...
+    if oc get subscription openshift-pipelines-operator-rh -n openshift-operators &>/dev/null; then
+        echo "Uninstalling OpenShift Pipelines Operator"
+        oc delete subscription openshift-pipelines-operator-rh -n openshift-operators
+        if oc get clusterserviceversion -n openshift-operators | grep openshift-pipelines-operator-rh &>/dev/null; then
+            oc delete clusterserviceversion $(oc get clusterserviceversion -n openshift-operators | grep openshift-pipelines-operator-rh | awk '{print $1}') -n openshift-operators
+        fi
+    else
+        echo "OpenShift Pipelines Operator not found"
+    fi
+
```
Comment (kpouget) [inline projects/notebooks/testing/prepare_sutest.sh:289]: this can be removed

File: projects/notebooks/testing/prepare_sutest.sh [hunk 9]
```diff
@@ -134,7 +142,7 @@ sutest_wait_rhods_launch() {
         ./run_toolbox.py from_config cluster set_project_annotation --prefix sutest --suffix toleration
     fi
 
-    ./run_toolbox.py rhods update_datasciencecluster --enable [dashboard,workbenches]
+    ./run_toolbox.py rhods update_datasciencecluster --enable [dashboard,workbenches,datasciencepipelines]
```
Comment (kpouget) [inline projects/notebooks/testing/prepare_sutest.sh:145]: can you change this to something like that:
```suggestion
to_enable=dashboard,workbenches
if [[ $(get_config tests.notebooks.test_flavor) == dashboard-scale-test ]]; then
    to_enable="${to_enable},datasciencepipelines"
fi
./run_toolbox.py rhods update_datasciencecluster --enable "[$to_enable]"
```

**Comment by smandaRH** [comment]
I will update the reviewed request changes.

File: projects/notebooks/testing/prepare_sutest.sh [hunk 10]
```diff
@@ -74,6 +73,7 @@ prepare_ocp_sutest_deploy_rhods() {
         oc adm groups new dedicated-admins
         oc adm policy add-cluster-role-to-group cluster-admin dedicated-admins
     fi
+
```
Comment (kpouget) [inline projects/notebooks/testing/prepare_sutest.sh:76]: no need for this new line, nor for the one (removed) the beginning of the file

File: projects/notebooks/testing/prepare_sutest.sh [hunk 11]
```diff
@@ -134,7 +134,11 @@ sutest_wait_rhods_launch() {
         ./run_toolbox.py from_config cluster set_project_annotation --prefix sutest --suffix toleration
     fi
 
-    ./run_toolbox.py rhods update_datasciencecluster --enable [dashboard,workbenches]
+    to_enable=dashboard,workbenches
+    if [[ $(get_config tests.notebooks.test_flavor) == dashboard-scale-test ]]; then
+        to_enable="${to_enable},datasciencepipelines"
+    fi
+./run_toolbox.py rhods update_datasciencecluster --enable "[$to_enable]"
```
Comment (kpouget) [inline projects/notebooks/testing/prepare_sutest.sh:141]:
```suggestion
    ./run_toolbox.py rhods update_datasciencecluster --enable "[$to_enable]"
```

File: projects/notebooks/testing/config.yaml [hunk 12]
```diff
@@ -257,12 +269,12 @@ tests:
...
     test_flavor: ods-ci
     identifier: unspecified
-    minio_bucket_name: notebooks
+    minio_bucket_name: ds-pipeline-dspa
     # number of test repetition to perform
     repeat: 1
     users:
       # number of users to simulate
-      count: 4
+      count: 5
```
Comment (kpouget) [inline projects/notebooks/testing/config.yaml:277]: this, and below, and above (dedicated) shouldn't be changed, so that we can merge the PR

File: projects/notebooks/testing/config.yaml [hunk 13]
```diff
@@ -66,7 +66,19 @@ ci_presets:
     tests.notebook_performance.test_only_one_image: true
 
   dashboard_scale_test:
```
Comment (kpouget) [inline projects/notebooks/testing/config.yaml:68]: we'll need to rediscuss this one to agree on the properties

File: projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml [hunk 14]
```diff
@@ -199,7 +251,7 @@
   until: wait_rhods_test_job_start.stdout
 
 - name: Wait for the RHODS tester job to terminate
-  command:
+  command:
```
Comment (kpouget) [inline projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml:254]: can you remove the trailing space?

Comment (kpouget) [reply projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml:254]: (everywhere where the linter is complaining)

File: projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml [hunk 15]
```diff
@@ -45,17 +51,34 @@
...
-      set -o pipefail;
-      oc get csv -oname -n redhat-ods-operator | grep rhods-operator
+    shell: set -o pipefail; oc get csv -oname -n redhat-ods-operator | grep rhods-operator
     register: rhods_csv_cmd
 
   - name: Get the RHODS version
-    shell:
-      set -o pipefail;
-      oc get {{ rhods_csv_cmd.stdout }} -n redhat-ods-operator -oname | grep rhods-operator | cut -d/ -f2 | cut -d. -f2-
+    shell: set -o pipefail; oc get {{ rhods_csv_cmd.stdout }} -n redhat-ods-operator -oname | grep rhods-operator | cut -d/ -f2 | cut -d. -f2-
```
Comment (kpouget) [inline projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml:58]: can you revert this change, and the one above?

File: projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml [hunk 16]
```diff
@@ -27,7 +26,14 @@
...
     capture_prom_db: "{{ notebooks_dashboard_scale_test_capture_prom_db }}"
 
+- name: Set system-under-test == driver-cluster if no system-under-test (SUT) is provided
+  set_fact:
+    sut_cluster_kubeconfig: "{{ notebooks_dashboard_scale_test_sut_cluster_kubeconfig | default(lookup('env', 'KUBECONFIG'), true) }}"
+
+
 - name: Fetch RHODS endpoints from the SUT cluster
+  environment:
+    KUBECONFIG: '{{ sut_cluster_kubeconfig }}'
```
Comment (kpouget) [inline projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml:36]: you don't need this kubeconfig machinery, I think I removed it all when I prepared the ansible role for you

File: projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml [hunk 17]
```diff
@@ -1,4 +1,3 @@
----
```
Comment (kpouget) [inline projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml:1]: shouldn't be removed

File: projects/notebooks/roles/notebooks_dashboard_scale_test/files/ods-ci/notebook_dsg_test.robot [hunk 18]
```diff
@@ -3,6 +3,9 @@ Resource            tests/Resources/ODS.robot
 Resource            tests/Resources/Common.robot
```
Comment (kpouget) [inline projects/notebooks/roles/notebooks_dashboard_scale_test/files/ods-ci/notebook_dsg_test.robot:3]: maybe you can rename this file to something like dashboard elyra scale test ?

File: projects/notebooks/roles/notebooks_dashboard_scale_test/files/entrypoint.sh [hunk 19]
```diff
@@ -91,5 +95,6 @@ echo "$test_exit_code" > "${ARTIFACT_DIR}/test.exit_code"
 
 echo "Test finished with $test_exit_code errors."
 echo "test_execution: $(date)" >> "${ARTIFACT_DIR}/progress_ts.yaml"
+rm ${ARTIFACT_DIR}/Standard Data Science Pipeline-*
```
Comment (kpouget) [inline projects/notebooks/roles/notebooks_dashboard_scale_test/files/entrypoint.sh:98]: I'm not sure about this one. The spaces won't work, and it's directories

I think this should be in the ansible role, not in this script

File: projects/notebooks/roles/notebooks_dashboard_scale_test/files/entrypoint.sh [hunk 20]
```diff
@@ -91,5 +95,6 @@ echo "$test_exit_code" > "${ARTIFACT_DIR}/test.exit_code"
...
 
-exit 0 # always exit 0, we'll decide later if this is a success or a failure
+exit 0 # always exit 0, we'll decide later if this is a success or a failure
```
Comment (kpouget) [inline projects/notebooks/roles/notebooks_dashboard_scale_test/files/entrypoint.sh:100]: can you add a new line delimiter at the end of the file?

File: projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml [hunk 21]
```diff
@@ -117,6 +140,22 @@
...
+
+  - name: Set s3_host_base fact
+    set_fact:
+      s3_host_base: "{{ hostname_result.stdout }}"
+
+  - name: Debug the s3_host_base variable (optional)
+    debug:
+      msg: "s3_host_base is set to {{ s3_host_base }}"
+
+
```
Comment (kpouget) [inline projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml:158]: you can remove these changes. The hostname is visible in the logs of the command above

File: projects/notebooks/roles/notebooks_dashboard_scale_test/templates/ods-ci_job.yaml [hunk 22]
```diff
@@ -51,6 +51,26 @@ spec:
...
+          value: "{{ rhods_dashboard_product_name }}"
+        - name: S3_BUCKET_NAME
+          value: "{{ notebooks_dashboard_scale_test_minio_bucket_name }}"
+        - name: MINIO_NAMESPACE
+          value: "{{ notebooks_dashboard_scale_test_minio_namespace }}"
+        - name: S3_SECRET_KEY
+          valueFrom:
+            secretKeyRef:
+              name: minio
+              key: MINIO_S3_SECRET_KEY
```
Comment (kpouget) [inline projects/notebooks/roles/notebooks_dashboard_scale_test/templates/ods-ci_job.yaml:64]: I don't really like this, if in the script we put the `env` command, the secret key is leaked :/ 
We'll need to rediscuss this

Comment (kpouget) [reply projects/notebooks/roles/notebooks_dashboard_scale_test/templates/ods-ci_job.yaml:64]: should be passed within the variables.yaml something file, which already handles secret values

Comment (smandaRH) [reply projects/notebooks/roles/notebooks_dashboard_scale_test/templates/ods-ci_job.yaml]: understood,  removed secret from the env and updated the secret with TESTUSER password.

File: projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml [hunk 23]
```diff
@@ -27,6 +27,11 @@
     test_artifacts_collected: "{{ notebooks_dashboard_scale_test_artifacts_collected }}"
     capture_prom_db: "{{ notebooks_dashboard_scale_test_capture_prom_db }}"
 
+- name: Set system-under-test == driver-cluster if no system-under-test (SUT) is provided
+  set_fact:
+    sut_cluster_kubeconfig: "{{ notebooks_dashboard_scale_test_sut_cluster_kubeconfig | default(lookup('env', 'KUBECONFIG'), true) }}"
+
+
```
Comment (kpouget) [inline projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml:34]: you need to remove this as well

File: projects/notebooks/testing/config.yaml [hunk 24]
```diff
@@ -257,7 +259,7 @@ tests:
     # The test flavor to run. Can be: ods-ci, notebook-performance or gating
     test_flavor: ods-ci
     identifier: unspecified
-    minio_bucket_name: notebooks
+    minio_bucket_name: ds-pipeline-dspa
```
Comment (kpouget) [inline projects/notebooks/testing/config.yaml:262]: you need to revert this modification

File: projects/notebooks/testing/config.yaml [hunk 25]
```diff
@@ -283,7 +285,7 @@ tests:
     artifacts_collected: no-screenshot-except-failed-and-zero
     # Should be true to capture Prometheus databases and properly generate the cluster usage metrics
     # Can be overriden by CI_ARTIFACTS_CAPTURE_PROM_DB
-    capture_prom_db: true
+    capture_prom_db: false
```
Comment (kpouget) [inline projects/notebooks/testing/config.yaml:288]: you need to revert this modification (and maybe include it in the `dashboard_scale_test` preset)

File: projects/notebooks/roles/notebooks_dashboard_scale_test/templates/ods-ci_job.yaml [hunk 26]
```diff
@@ -51,6 +51,21 @@ spec:
...
+          value: "{{ notebooks_dashboard_scale_test_minio_bucket_name }}"
+        - name: MINIO_NAMESPACE
+          value: "{{ notebooks_dashboard_scale_test_minio_namespace }}"
+        - name: DASHBOARD_PRODUCT_NAME
+          value: "{{ rhods_dashboard_product_name }}"
+        - name: S3_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              name: minio
+              key: MINIO_S3_ACCESS_KEY
```
Comment (kpouget) [inline projects/notebooks/roles/notebooks_dashboard_scale_test/templates/ods-ci_job.yaml:64]: can you pass the access key via the ODS-CI config file? it's not exactly secret, but better keep it hidden

File: projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/artifacts_driver.yml [hunk 27]
```diff
@@ -59,7 +59,9 @@
     oc -n {{ notebooks_dashboard_scale_test_minio_namespace }} -c ubi8 \
        rsync "{{ minio_podname_cmd.stdout }}:/artifacts/to_export/" "{{ artifact_extra_logs_dir }}" --quiet
 
-    rm -f "{{ artifact_extra_logs_dir }}/os-release"
+    rm -f "{{ artifact_extra_logs_dir }}/os-release" \
+    find "{{ artifact_extra_logs_dir }}/" -type d -name "Standard Data Science Pipeline*" -exec rm -rf {} + \
+    rm -rf "{{ artifact_extra_logs_dir }}/pipelines"
```
Comment (kpouget) [inline projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/artifacts_driver.yml:64]: you need to remove the `\` at the end of the string, it won't work with it I think

File: projects/notebooks/testing/config.yaml (hunk 24 above)
Comment (smandaRH) [reply projects/notebooks/testing/config.yaml:262]: while  RUNNING THIS STEP it is trying to emty notebooks bucket instead of ds-pipeline-dspa  bucket
```shell

``` ~~ projects/notebooks/roles/notebooks_dashboard_scale_test/tasks/main.yml:174
~~ TASK: notebooks_dashboard_scale_test : Empty the Minio S3 bucket
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


<command>
oc -c mc -n notebook-scale-test-minio rsh $(oc get pod -lapp=minio -n notebook-scale-test-minio -oname)   mc --config-dir /tmp rm --insecure --recursive --force --quiet minio/ds-pipeline-dspa/ >/dev/null;
oc -c mc -n notebook-scale-test-minio rsh $(oc get pod -lapp=minio -n notebook-scale-test-minio -oname)   rm -rf /artifacts/to_export > /dev/null
oc -c mc -n notebook-scale-test-minio rsh $(oc get pod -lapp=minio -n notebook-scale-test-minio -oname)   mc --config-dir /tmp cp --insecure /etc/os-release minio/ds-pipeline-dspa; # without it, cp may fail if the bucket is empty

</command>

<stdout> `/etc/os-release` -> `minio/ds-pipeline-dspa/os-release`
<stdout> Total: 0 B, Transferred: 592 B, Speed: 109.51 KiB/s

Comment (kpouget) [reply projects/notebooks/testing/config.yaml:262]: yes, when you run it locally, the preset isn't applied.
so you need to remove it from the commit, but keep it applied in your machine if you want to test locally

**Comment by smandaRH** [comment]
Running test to check whether updated test cases are working fine or not.

File: projects/notebooks/testing/config.yaml [hunk 28]
```diff
@@ -213,10 +215,10 @@ ldap:
 
 rhods:
   catalog:
-    image: brew.registry.redhat.io/rh-osbs/iib
-    tag: 752361
+    image: registry-proxy.engineering.redhat.com/rh-osbs/iib
```
Comment (kpouget) [inline projects/notebooks/testing/config.yaml:218]: you need to change this back to `brew.registry.redhat.io/rh-osbs/iib`, this one won't work

```suggestion
    image: brew.registry.redhat.io/rh-osbs/iib
```

**Comment by kpouget** [comment]
oh, that's [something](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/407/pull-ci-openshift-psap-topsail-main-rhoai-light/1816303890673438720/artifacts/light/test/artifacts/002__driver_notebooks__dashboard_scale_test/) !
image

**Comment by kpouget** [comment]
very good job @smandaRH , I merge this one!
